### PR TITLE
Added libmustache in prep for adding php55-mustache.

### PR DESCRIPTION
**Pull request by Stubbs**
Added libmustache in prep for adding php55-mustache, the PHP Mustache template extension.

### CODE REVIEW

File: Library/Formula/libmustache.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  def install
+    system "autoreconf", "-fvi"
+    system "./configure", "--disable-debug",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/mustache", "-h"
```
Comment (mikemcquaid) [inline Library/Formula/libmustache.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by Stubbs** [comment]
With a little help from the upstream author I've added a small test source file that gets compiled, linked & executed, pretty similar to tinyxml2 ...

**Comment by Stubbs** [comment]
Not really sure what to do about this last failure, when I add a "desc" field to the formula my audit throws an error `Error: undefined method `desc' for Formulary::Formulae::Libmustache:Class`, same if I use `description`.

**Comment by DomT4** [comment]
Make sure your Homebrew is up-to-date and this branch is up-to-date with the Master branch.

**Comment by Stubbs** [comment]
I thought I'd done that, but I hadn't then pulled those changes into my libmustache branch. I've rebased my branch and it's all gravy.

**Comment by mikemcquaid** [comment]
@BrewTestBot test this please

**Comment by Stubbs** [comment]
Is there anything else I need to do on this PR?

**Comment by DomT4** [comment]
My concern was that it's a little niche and may be better in the `php` tap, but if another @Homebrew/owners signs off on it I'm happy to pull.

**Comment by jacknagel** [comment]
True, the README does say "C++ implementation of Mustache intended mainly for use as a PHP extension".

**Comment by DomT4** [comment]
@Homebrew/php Mind accepting this in `Homebrew/php` instead of the core?

**Comment by alanthing** [comment]
@DomT4 If it's something that would generally only be used alongside the PHP extension and nothing else, sure.

**Comment by DomT4** [comment]
@alanthing Per Jack's comment just above it seems to be intended by the developer to be useful for PHP. If someone can provide a compelling test-case of "I will use it on OSX for xyz that isn't PHP" we could consider it, but otherwise I think it's too niche for the core personally.

**Comment by alanthing** [comment]
@DomT4 sounds great.

@Stubbs could you put this same formula in a PR for [Homebrew/homebrew-php](https://github.com/Homebrew/homebrew-php) ?

**Comment by DomT4** [comment]
Thanks @alanthing, Appreciated. Closing here.

**Pull request closed without merging**